sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultSettingsAsyncClient.java

```diff
@@ -0,0 +1,230 @@
+     * @throws KeyVaultErrorException thrown if the request is rejected by the server.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultSetting> getSetting(String name) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'name' cannot be empty or null"));
+        }
+
+        try {
+            return this.implClient.getSettingAsync(vaultUrl, name)
+                .doOnRequest(ignored -> logger.verbose("Retrieving account setting - {}", name))
+                .doOnSuccess(response -> logger.verbose("Retrieved account setting - {}", name))
+                .doOnError(error -> logger.warning("Failed retrieving account setting - {}", name, error))
+                .onErrorMap(KeyVaultAdministrationUtils::mapThrowableToKeyVaultAdministrationException)
+                .map(KeyVaultSettingsAsyncClient::transformToKeyVaultSetting);
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * Get the value of a specific account setting.
+     *
+     * @param name The name of setting to retrieve the value of.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * {@link KeyVaultSetting account setting}.
+     *
+     * @throws IllegalArgumentException thrown if {@code name} is {@code null} or empty.
+     * @throws KeyVaultErrorException thrown if the request is rejected by the server.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<KeyVaultSetting>> getSettingWithResponse(String name) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'name' cannot be empty or null"));
+        }
+
+        try {
+            return this.implClient.getSettingWithResponseAsync(vaultUrl, name)
+                .doOnRequest(ignored -> logger.verbose("Retrieving account setting - {}", name))
+                .doOnSuccess(response -> logger.verbose("Retrieved account setting - {}", name))
+                .doOnError(error -> logger.warning("Failed retrieving account setting - {}", name, error))
+                .onErrorMap(KeyVaultAdministrationUtils::mapThrowableToKeyVaultAdministrationException)
+                .map(response -> new SimpleResponse<>(response, transformToKeyVaultSetting(response.getValue())));
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * List the account's settings.
+     *
+     * @return A {@link Mono} containing the list of {@link KeyVaultSetting account settings}.
+     *
+     * @throws KeyVaultErrorException thrown if the request is rejected by the server.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Mono<List<KeyVaultSetting>> listSettings() {
```

Comment:
Talking with our architects, we should also return a list result much like the model type in case properties are added. We do that in REST for that reason - it'd be a breaking change - but we could also ask the team if they ever expect to add anything more to it.

Reply:
Sounds good, we will return a result object containing the list then.